### PR TITLE
feat/add_gui_for_SDFq

### PR DESCRIPTION
**Pull request by FloydZ**

### CODE REVIEW

**Review by Javierverbel — DISMISSED** [review]

File: input_dictionary.json [hunk 1]
```diff
@@ -461,6 +461,119 @@
...
+          "direction": "column",
+          "display_label": "Included algorithms",
+          "tooltip": "Algorithms to include for optimization",
+          "default_value": [],
+          "excluded_algorithms": [],
+          "options": [],
+          "dependencies": []
+        },
+        {
+          "id": "memory_bound",
```
Comment (Memphisd) [inline input_dictionary.json:539]: we said this s a problem parameter or not?

File: input_dictionary.json [hunk 2]
```diff
@@ -461,6 +461,119 @@
...
+          "direction": "column",
+          "display_label": "Decimal precision",
+          "default_value": 0,
+          "placeholder": "Insert value",
+          "tooltip": "Number of decimal digits to display"
+        }
+      ],
+      "optional_parameters": [
+        {
+          "id": "nsolutions",
```
Comment (Memphisd) [inline input_dictionary.json:569]: Same here, this is a problem parameter

**Review by Memphisd — CHANGES_REQUESTED** [review]
restructure parmeters

File: input_dictionary.json [hunk 3]
```diff
@@ -461,6 +461,120 @@
...
+          "type": "number",
+          "direction": "column",
+          "display_label": "Number of solutions",
+          "placeholder": "Insert value",
+          "tooltip": "Log2 of number of existing solutions of which one has to be found, if not specified it will be fixed to the number of solutions in expectation"
+        }
+      ],
+      "optional_parameters": [
+
```
Comment (Memphisd) [inline input_dictionary.json:576]: here should be the is_syndrome_zero parameter, named "Code word search" as a selector

File: input_dictionary.json [hunk 4]
```diff
@@ -461,6 +461,120 @@
...
+          "direction": "column",
+          "display_label": "Included algorithms",
+          "tooltip": "Algorithms to include for optimization",
+          "default_value": [],
+          "excluded_algorithms": [],
+          "options": [],
+          "dependencies": []
+        },
+        {
+          "id": "memory_bound",
```
Comment (Memphisd) [inline input_dictionary.json:539]: is a problem parameter

File: input_dictionary.json [hunk 5]
```diff
@@ -461,6 +461,120 @@
...
+          "id": "precision",
+          "type": "number",
+          "direction": "column",
+          "display_label": "Decimal precision",
+          "default_value": 0,
+          "placeholder": "Insert value",
+          "tooltip": "Number of decimal digits to display"
+        },
+        {
+          "id": "nsolutions",
```
Comment (Memphisd) [inline input_dictionary.json:567]: is a problem parameter

**Review by Memphisd — CHANGES_REQUESTED** [review]

**Review by Memphisd — APPROVED** [review]

**Pull request merged**